inventory/classes/components/rabbitmq-operator/rabbitmq-configuration.yml

```diff
@@ -0,0 +1,265 @@
+# This class contains example CRDs for the rabbitmq-operator to configure the rabbitmq cluster including users, vhosts, exchanges, policy etc...
+# generally all "components" accepts multiple entries e.g. multiple users
+# examples where taken from https://www.rabbitmq.com/kubernetes/operator/using-operator.html and https://www.rabbitmq.com/kubernetes/operator/using-topology-operator.html
+
+parameters:
+  rabbitmqcluster:
+    rabbitmqcluster:
+      annotations:
+        additional.annotation: rabbitmq
+      labels:
+        additional.label: rabbitmq
+      replicas: 3
+      image: my-private-registry/rabbitmq:my-custom-tag
```

Comment:
I can't help notice lots of repetitions from the generic kubernetes generator. I wonder if part of this could be done by the normal generator, in order to reduce duplication?

Reply:
@ademariag we'd like to, could you give us an idea how that can/should work for kapitan-reference?

Reply:
It might be possible, but could lead to more complexity in the kubernetes generator. The rabbitmq-operator requires an CRD that looks like a normal Kubernetes deployment/statefulset but it is not. The CRD has some additional fields and is also using a different apiVersion. 
Take a look at [this](compiled/rabbitmq-operator/rabbitmq/rabbitmqcluster-rabbitmq.yml) compiled CRD and also [this](https://github.com/rabbitmq/cluster-operator/blob/main/docs/examples/production-ready/rabbitmq.yaml) example from the official rabbitmq-operator.